src/Analyser/TypeSpecifier.php

```diff
@@ -880,8 +882,19 @@ public function specifyTypesInCondition(
 		} elseif (
 			$expr instanceof Expr\Isset_
 			&& count($expr->vars) > 0
-			&& $context->true()
+			&& !$context->null()
 		) {
+			if (!$context->true()) {
+				return array_reduce(
+					array_filter(
+						$expr->vars,
+						static fn (Expr $var) => $var instanceof Expr\Variable && is_string($var->name) && $scope->hasVariableType($var->name)->yes(),
+					),
+					fn (SpecifiedTypes $types, Expr\Variable $var) => $types->unionWith($this->specifyTypesInCondition($scope, $var, $context, $rootExpr)),
```

Comment:
Looks like this will just affectively call `handleDefaultTruthyOrFalseyContext()` which I could do directly too then I guess. Maybe I should set the type to `StaticTypeFactory::falsey()` minus null actually. I'll need to check this again

Reply:
should be fine, I prefer the more generic approach. it should be possible to re-use this somehow for ArrayDimFetch and PropertyFetch too